src/js/libgeo/GeoOps.js

```diff
@@ -616,15 +644,170 @@ geoOps.PointOnSegment.updatePosition = function(el) {
     var param = getStateComplexNumber();
     putStateComplexNumber(param); // copy parameter
     var seg = csgeo.csnames[el.args[0]];
-    var start = seg.startpos;
-    var end = seg.endpos;
-    var far = List.sub(end, start);
-    var homog = List.add(start, List.scalmult(param, far));
-    homog = List.normalizeMax(homog);
+
+    var homog = geoOps._helper.projectPointToSegment(seg, param);
     el.homog = General.withUsage(homog, "Point");
 };
 geoOps.PointOnSegment.stateSize = 2;
 
+geoOps._helper.PointOnArcCr = function(arc, P) {
+    var A = arc.startPoint,
+        B = arc.viaPoint,
+        C = arc.endPoint;
```

Comment:
Minor nit: I prefer repeating the `var` keyword. Do we want to achieve consistent styling on this?